### PR TITLE
redoes construct wall phasing

### PR DESCRIPTION
**Pull request by ShadowDeath6**
no more ugly phasing, it's kinda glitchy
also fixes an infinite runed metal exploit, don't tell nobody

:cl: ShadowDeath6
tweak: Constructs don't make walls phasable anymore, they just go through them like shadowlings.
bugfix: Constructs can't pull things through walls anymore.
/:cl:

### CODE REVIEW

**Comment by Super3222** [comment]
> bugfix: Constructs can't pull things through walls anymore.

I thought I fixed this?

**Comment by ShadowDeath6** [comment]
they could pull shit through in between the ticks

**Review by X-TheDark — APPROVED** [review]

**Comment by X-TheDark** [comment]
This will also make the artificers move through walls faster, won't it?

**Comment by ShadowDeath6** [comment]
The only reason they move faster is because they don't get stopped on walls anymore, so I guess you could say that

File: code/game/turfs/simulated/walls.dm [hunk 1]
```diff
@@ -102,9 +102,9 @@
 
 /turf/closed/wall/attack_animal(mob/living/simple_animal/M)
 	if(istype(M,/mob/living/simple_animal/hostile/construct/builder)||istype(M,/mob/living/simple_animal/hostile/construct/harvester))
-		if(istype(src, /turf/closed/wall/mineral/cult))
+		if(istype(src, /turf/closed/wall/mineral/cult)||istype(src, /turf/closed/wall/mineral/cult/artificer))
```
Comment (Super3222) [inline code/game/turfs/simulated/walls.dm]: erm why.

Comment (X-TheDark) [reply code/game/turfs/simulated/walls.dm]: For some reason I thought it tried to check the type of mob...by bad for missing this.

Yeah, this isn't necessary, `/turf/closed/wall/mineral/cult/artificer` already passes the `if(istype(src, /turf/closed/wall/mineral/cult)` check.

Comment (ShadowDeath6) [reply code/game/turfs/simulated/walls.dm]: I noticed this a while ago, I just forgot to fix it, wew.

**Pull request merged**